zhmccli/_cmd_partition.py

```diff
@@ -916,6 +967,133 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
+            uuid = volume_parts[0]
+            sg_list = partition.list_attached_storage_groups()
+            storage_volume = None
+            for sg in sg_list:
+                try:
+                    storage_volume = sg.storage_volumes.find(uuid=uuid)
+                    break
+                except zhmcclient.NotFound:
+                    continue
+            if not storage_volume:
+                raise_click_exception(
+                    "Storage volume with UUID '{uuid}' specified in the "
+                    "--volume option was not found in any storage group "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(uuid=uuid, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+        elif len(volume_parts) == 2:  # format: SG.SV
+            sg_name, sv_name = volume_parts
+            sg_list = partition.list_attached_storage_groups()
+            sg = None
+            for _sg in sg_list:
+                if _sg.name == sg_name:
+                    sg = _sg
+                    break
+            if not sg:
+                raise_click_exception(
+                    "Storage group '{sg}' specified in the --volume option was "
+                    "not found in the storage groups attached to partition "
+                    "'{part}' of CPC '{cpc}'.".
+                    format(sg=sg_name, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+            try:
+                storage_volume = sg.storage_volumes.find(name=sv_name)
+            except zhmcclient.NotFound:
+                raise_click_exception(
+                    "Storage volume '{sv}' specified in the --volume option "
+                    "was not found in the volumes of storage group '{sg}' "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(sv=sv_name, sg=sg_name, part=partition_name,
+                           cpc=cpc_name),
+                    cmd_ctx.error_format)
+        else:
+            raise_click_exception(
+                "Invalid format for --volume option: '{v}'. CPC '{cpc}' has "
+                "the storage management feature which requires the formats "
+                "'SG.SV' or 'UUID'.".
+                format(v=volume_opt, cpc=cpc_name),
+                cmd_ctx.error_format)
+
+        # Set the remaining feature-specific parameters
+        properties['storage-volume-uri'] = storage_volume.uri
+        properties['timeout'] = timeout
+
+        op_properties = {
+            'dump-program-info': properties,
+            'dump-program-type': 'storage',
+        }
+        try:
+            partition.start_dump_program(
+                parameters=op_properties, wait_for_completion=True)
+        except zhmcclient.Error as exc:
+            raise_click_exception(exc, cmd_ctx.error_format)
+
+    else:  # No storage management feature
+
```

Comment:
I would probably factor out the with-storage-management-dump and the without-storage-management-dump code into two separate functions, otherwise this function is pretty long and harder to test.

```
if storage_management_feature(partition):
    dump_....(...)
else:
    dump_....(...)
```

Reply:
I like that idea and have split it up. The new code also no longer needs to make the assumption about commonality of the input parameters between "Dump Partition" and "Start Dump Program".
DONE